### PR TITLE
Use explicit variable types to support different vars usecases

### PR DESCRIPTION
**Pull request by ryanwholey**
Adds interim types (`action.Variables \ action.VariablesItem)`) that will act as a transitional type between variable input types (`action.VariablesInput \ action.VariablesInputItem`) and the Terraform variable resource type (`tfeprovider.Variable`).

This allows to have explicit types for the different use cases throughout the action:
- `action.VariablesInput` is used specifically to unmarshal the GitHub action input
- `action.Variables` stores the workspace reference in addition to the variable attributes, which allows us to pass this struct around to import the correct variables as well as hold reference to the workspace fields needed to create the `tfeprovider.Variable` struct
- `tfeprovider.Variable` is the resource that will be marshaled into json for Terraform consumption, and therefor should be the type that stores Terraform specific logic (for_each, workspace_id, etc)

### CODE REVIEW

File: internal/action/variable.go [hunk 1]
```diff
@@ -10,7 +16,53 @@ type VariablesInputItem struct {
 	Description string `yaml:"description,omitempty"`
 	Category    string `yaml:"category,omitempty"`
 	Sensitive   bool   `yaml:"sensitive,omitempty"`
+}
+
+type Variables []VariablesItem
+
+type VariablesItem struct {
```
Comment (bendrucker) [inline internal/action/variable.go:23]: Can probably just be `Variable` no?

File: internal/action/variable.go [hunk 2]
```diff
@@ -10,7 +16,53 @@ type VariablesInputItem struct {
...
+		Key:         vi.Key,
+		Value:       vi.Value,
+		Description: vi.Description,
+		Category:    vi.Category,
+		Sensitive:   vi.Sensitive,
+		Workspace:   w,
+	}
+}
+
+func (avi VariablesItem) ToResource() *tfeprovider.Variable {
```
Comment (bendrucker) [inline internal/action/variable.go:43]: Stick to the same receiver name

**Review by bendrucker — APPROVED** [review]
Nice!

File: internal/action/variable.go (hunk 2 above)
Comment (ryanwholey) [reply internal/action/variable.go:43]: ah whoops, left over after a naming refactor (my first idea was ActionVariables which is not a good name)

**Review by bendrucker — APPROVED** [review]

**Pull request merged**